app/assets/javascripts/discourse/app/lib/tracked-tools.js

```diff
@@ -176,3 +176,65 @@ export class DeferredTrackedSet {
     next(() => this.#set.clear());
   }
 }
+
+/**
+ * @decorator
+ *
+ * Same as @tracked, but initializes the value as a TrackedArray.
+ *
+ * @param {Object} target - The target object
+ * @param {string|Symbol} key - The property key
+ * @param {Object} desc - The property descriptor
+ * @throws {Error} If value is not an array, TrackedArray, or null
+ *
+ * @returns {Object} Modified property descriptor that wraps arrays in TrackedArray
+ *
+ * @example
+ * class TodoList {
+ *   @trackedArray todos = ['Buy milk', 'Walk dog'];
+ *
+ *   addTodo(text) {
+ *     // Automatically wrapped in TrackedArray
+ *     this.todos = [...this.todos, text];
+ *   }
+ * }
+ */
+export function trackedArray(target, key, desc) {
+  if (desc.initializer) {
+    const initialValue = desc.initializer();
```

Comment:
I think it's too early to evaluate the initializer here. It should be executed once-per-instance, and it should have access to the instance in `this.`. So it should be more like 

```
const originalInitializer = desc.initializer;
desc.initializer = function() {
  const initialValue = desc.initializer.apply(this);

  // Do your TrackedArray conversion logic here
}
```